cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/instance/AwsNativeEIPResourceBuilder.java

```diff
@@ -148,13 +152,15 @@ public CloudResource delete(AwsContext context, AuthenticatedContext auth, Cloud
             DisassociateAddressRequest disassociateAddressRequest = new DisassociateAddressRequest()
                     .withAssociationId(eipAttributes.getAssociationId());
             awsMethodExecutor.execute(() -> context.getAmazonEc2Client().disassociateAddress(disassociateAddressRequest), null);
-            ReleaseAddressRequest request = new ReleaseAddressRequest()
-                    .withAllocationId(eipAttributes.getAllocateId());
-            ReleaseAddressResult releaseAddressResult = awsMethodExecutor.execute(() -> context.getAmazonEc2Client().releaseAddress(request), null);
-            ret = releaseAddressResult == null ? null : resource;
         } else {
-            LOGGER.debug("Cannot find attribute for {}, release and disassociate skipped", resource.getName());
+            LOGGER.info("Cannot find attribute for {}, disassociate address operation is skipped", resource.getName());
         }
+        String allocationId = resource.getReference();
+        LOGGER.debug("Releasing EIP address with allocation id: '{}'", allocationId);
+        ReleaseAddressRequest request = new ReleaseAddressRequest()
+                .withAllocationId(allocationId);
+        ReleaseAddressResult releaseAddressResult = awsMethodExecutor.execute(() -> context.getAmazonEc2Client().releaseAddress(request), null);
+        ret = releaseAddressResult == null ? null : resource;
         return ret;
```

Comment:
I think `ret` can be eliminated (see also line 150) and joined with the `return` statement.